app/src/main/java/com/lubenard/oring_reminder/DbManager.java

```diff
@@ -237,6 +237,46 @@ public void endSession(long entryId) {
         if (entryId < 0)
             return;
 
+        // First we catch the dateTablePut date
+        String[] columns = new String[]{ringTablePut,ringTableRemoved};
+        Cursor cursor = readableDB.query(ringTable, columns,ringTableId + "=?",
+                new String[]{String.valueOf(entryId)}, null, null, null);
+
+        cursor.moveToFirst();
+        // Then we set our values:
+        // We need to recompute the date
+        // And set the isRunning to 0
+        String dateRemoved = "";
```

Comment:
I missed that the method dbManager.updateDatesRing is already used to update the end & start date.
so maybe it just need to have a condition on the dbManager.endSession to check if there is already a dateRemoved on the database, if yes we use this date to compute the end session. If not, we retrieve the current date

Reply:
In the updateDateRing, the session is already set to finished (isRunning is 1 if the session is running, else 0)
We should just call endPause, because we already know the session has been set to finish. No need to go through that big function that endSession is

Reply:
Ok I get it now, yes much better. just need to change the endPause method to public as well, not sure if it's an issue

Reply:
Not a issue at all, by default all methods are private, just to help understanding the code better